src/types/router.ts

```diff
@@ -26,4 +26,16 @@ export type Router<
   forward: () => void,
   go: (delta: number) => void,
   install: (app: App) => void,
+}
+
+export type RouterImplementation = {
+  routes: RouteMethodsImplementation,
+  route: DeepReadonly<Resolved<Route>>,
+  resolve: RouterResolve,
+  push: RouterPushImplementation,
+  replace: RouterReplace,
+  back: () => void,
+  forward: () => void,
+  go: (delta: number) => void,
+  install: (app: App) => void,
```

Comment:
I like the syntax of having all the "...Implementation" properties coming together here. Thoughts on following that  even further, maybe created Implementation types for resolve, replace? Just thinking out loud, def not a blocker

Reply:
We could, but they'd be the exact same types. Those types don't have any generics as is.